utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -400,33 +401,76 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
         self.get(code_point).into()
     }
 
-    /// Returns a [`CodePointMapRange`] struct representing a range of code
-    /// points associated with the same value.
-    pub fn get_range(&self, start: u32) -> Option<CodePointMapRange> {
+    /// Returns a [`CodePointMapRange`] struct which represents a range of code
+    /// points associated with the same trie value. The returned range will be
+    /// the longest stretch of consecutive code points starting at `start` that
+    /// share this value.
+    ///
+    /// This method is designed to use the internal details of
+    /// the structure of [`CodePointTrie`] to be optimally efficient. This will
+    /// outperform a naive approach that just uses [`CodePointTrie::get()`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_codepointtrie::planes;
+    /// let trie = planes::get_planes_trie();
+    ///
+    /// pub const CODE_POINT_MAX: u32 = 0x10ffff;
+    ///
+    /// let start = 0x1_0000;
+    /// let exp_end = 0x1_ffff;
+    ///
+    /// let start_val = trie.get(start);
+    /// assert_eq!(trie.get(exp_end), start_val);
+    /// assert_ne!(trie.get(exp_end + 1), start_val);
+    ///
+    ///
+    /// // Slower naive approach using `CodePointTrie::get()`
+    ///
+    /// let start_to_max = (start..=CODE_POINT_MAX).into_iter();
```

Comment:
> This portion loops over plane 1 using `.get()` to show an equivalent computation to `.get_range()` that exemplifies conceptually in more detail what `.get_range()` accomplishes, while the docs/comments assert that `.get_range()` is more efficient.
> 
> Should I remove it?

In the documentation of .get_range() I find it unnecessary to show an example of using a loop with .get().
I would remove it, but I don't feel super strongly about it.

> Also, this seems like an idea to create a test_util method that works on the exported ICU trie test case data that verifies `.get_range()` solely using `.get()`, although it would incur some time to actually iterate through all 1.1M code points in each test file. Would that be desirable or not?

What I do in the C and Java tests is that I have a list of expected ranges, and I *separately* test getRange() vs. get() and their variations against the expected ranges. This lets me isolate which function is misbehaving, if any.
See https://github.com/unicode-org/icu/blob/main/icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/CodePointTrieTest.java#L237

Reply:
Done. Removed the other example.

The testing of get_range() and get() as you described sounds good to me, and is/was already done. This PR updates the `test/test_util.rs` module in `icu_codepointtrie` (which is used by `tests/trie_test_data_test.rs` to run tests over all of the trie test cases exported from ICU) to have both such tests directly testing against check_ranges. `test_util::check_trie()` existed before this PR and verifies the `get()` logic against check_ranges, while this PR adds `test_util::test_check_ranges_get_ranges()` to verify `get_range()`. We don't have any special case fast path versions of `get()`, so I think we have those tests accounted for appropriately for the code that we have. Admittedly, the non-starter proposed 3rd test described above that would have tested get_range() solely using get() would have been redundant since it depends on get() being correct, which is ultimately verified using check_ranges.